File: src/my_cpp_pkg/src/rover_sub.cpp
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
using std::placeholders::_1;

class RoverSubscriber : public rclcpp::Node
{
public:
  RoverSubscriber() : Node("rover_subscriber")
  {
    subscription_ = this->create_subscription<std_msgs::msg::String>(
      "ares_nova_telemetry", 10, std::bind(&RoverSubscriber::topic_callback, this, _1));
  }

private:
  void topic_callback(const std_msgs::msg::String & msg) const
  {
    RCLCPP_INFO(this->get_logger(), "Received Data: '%s'", msg.data.c_str());
  }
  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<RoverSubscriber>());
  rclcpp::shutdown();
  return 0;
}

File: Task_1_ROS2_Basics/src/my_python_pkg/my_python_pkg/rover_sub.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class RoverSubscriber(Node):
    def __init__(self):
        super().__init__('py_rover_subscriber')
        self.subscription = self.create_subscription(
            String,
            'ares_nova_telemetry',
            self.listener_callback,
            10)
        self.subscription  # prevent unused variable warning

    def listener_callback(self, msg):
        self.get_logger().info(f'Received Data: "{msg.data}"')

def main(args=None):
    rclpy.init(args=args)
    rover_subscriber = RoverSubscriber()
    rclpy.spin(rover_subscriber)
    rover_subscriber.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()